flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableDescriptor.java

```diff
@@ -355,6 +355,22 @@ public Builder distributedBy(TableDistribution distribution) {
             return this;
         }
 
+        public Builder distributedByHash(String... bucketKeys) {
+            return distributedByHash(null, bucketKeys);
+        }
+
+        public Builder distributedByHash(Integer number, String... bucketKeys) {
+            return distributedBy(TableDistribution.ofHash(Arrays.asList(bucketKeys), number));
+        }
+
+        public Builder distributedByRange(String... bucketKeys) {
+            return distributedByRange(null, bucketKeys);
+        }
+
+        public Builder distributedByRange(Integer number, String... bucketKeys) {
+            return distributedBy(TableDistribution.ofRange(Arrays.asList(bucketKeys), number));
+        }
```

Comment:
add distributedInto(int number) for the UNKNOWN case

Reply:
I added `public Builder distributedInto(int number, String... bucketKeys)`.

The varargs will do the right thing if called with no bucketKeys.  

I added tests and exceptions for when there's an algorithm and no bucket keys specified.  

This would be the same as rejecting `DISTRIBUTED BY HASH INTO n BUCKETS`.